src/signature.c

```diff
@@ -361,13 +387,13 @@ static bool validate_certificate(void)
 		X509_STORE_CTX_free(verify_ctx);
 	}
 
-	return false;
+	return verify_ctx->error;
```

Comment:
+1 on this, we print out the error in the callback so we should return it too for better debugging.